src/cpu/aarch64/jit_sve_512_x8s8s32x_conv_kernel.hpp

```diff
@@ -0,0 +1,296 @@
+
+    bool mask_gflag;
+
+    ZReg vmm_out(int i_ur, int i_oc) {
+        int nb_x_blocking
+                = jcp.is_depthwise ? jcp.nb_ch_blocking : jcp.nb_oc_blocking;
+        int idx = i_ur * nb_x_blocking + i_oc;
+        assert(idx < (jcp.is_depthwise
+                               ? ker_dw_reg_base_idx
+                               : jcp.src_zero_point ? ker_zp_reg_base_idx
+                                                    : ker_reg_base_idx));
+        return ZReg(idx);
+    }
+    ZReg zmm_out(int i_ur, int i_oc) {
+        int idx = vmm_out(i_ur, i_oc).getIdx();
+        assert(idx
+                < (jcp.is_depthwise ? ker_dw_reg_base_idx : ker_reg_base_idx));
+        return ZReg(idx);
+    }
+    ZReg vmm_inp(int i_ic, int nb_x_blocking) {
+        int idx = i_ic + nb_x_blocking * jcp.ur_w;
+        assert(idx < 31);
+        return ZReg(idx);
+    }
+    ZReg zmm_inp(int i_ic, int nb_x_blocking) {
+        int idx = i_ic + nb_x_blocking * jcp.ur_w;
+        const int max_idx = jcp.src_zero_point ? ker_zp_reg_base_idx
+                                               : ker_dw_reg_base_idx;
+        assert(idx < max_idx);
+        MAYBE_UNUSED(max_idx);
+
+        return ZReg(idx);
+    }
+    ZReg vmm_bias_alpha() {
+        int nb_c_block
+                = jcp.is_depthwise ? jcp.nb_ch_blocking : jcp.nb_oc_blocking;
+        return ZReg(nb_c_block * jcp.ur_w);
+    }
+    ZReg xmm_bias_alpha() {
+        int nb_c_block
+                = jcp.is_depthwise ? jcp.nb_ch_blocking : jcp.nb_oc_blocking;
+        return ZReg(nb_c_block * jcp.ur_w);
+    }
+    int get_ow_start(int ki, int pad_l) {
+        return nstl::max(0,
+                utils::div_up(pad_l - ki * (jcp.dilate_w + 1), jcp.stride_w));
+    }
+    int get_ow_end(int ur_w, int ki, int pad_r) {
+        return ur_w
+                - nstl::max(0,
+                        utils::div_up(
+                                pad_r - (jcp.kw - 1 - ki) * (jcp.dilate_w + 1),
+                                jcp.stride_w));
+    }
+
+    void prepare_output(int ur_w);
+    void store_output(int ur_w, bool last_oc_block_flag);
+    void compute_ker_dw(int ur_w, int pad_l, int pad_r,
+            ic_block_t last_ic_block_flag, bool h_padded);
+    void compute_ker(int ur_w, int pad_l, int pad_r,
+            ic_block_t last_ic_block_flag, bool h_padded = false);
+    void kh_loop(int ur_w, int pad_l, int pad_r, ic_block_t last_ic_block_flag);
+    void icb_loop(int ur_w, int pad_l, int pad_r, bool is_last_spatial_block);
+    void generate() override;
+    void cvt2ps(data_type_t type_in, ZReg ymm_in, const XReg reg_base,
+            const int offset, bool mask_flag);
+    // Vmm vmm_mask(const Vmm vmm_in, bool mask_flag, bool store = false);
+    void vmm_mask_all_one();
+    void vmm_load_src(ZReg src, XReg reg_addr, bool mask_flag);
+
+    int get_offset(int raw_offt) {
+
+        assert(raw_offt <= INT_MAX);
+        auto offt = static_cast<int>(raw_offt);
```

Comment:
nit: why do we need `static_cast<int>` since `raw_offt` is already of type `int` [maybe we need to change the type of `raw_offt` to `int64_t` ?] Also with above data types I think  `assert(raw_offt <= INT_MAX);` will always hold. 

Furthermore, [assuming `raw_offt` is defined to be of type `int64_t` do we also need `assert(raw_offt >= INT_MIN)`

Reply:
Thx! I'll delete it.